### PR TITLE
Add decfloat, float, boolean to interval conversions

### PR DESCRIPTION
**Pull request by sfc-gh-pbulawa**
## Add interval type conversion support for BOOLEAN and FLOAT types

This change implements conversion from Snowflake BOOLEAN and FLOAT types to ODBC interval C types (SQL_C_INTERVAL_YEAR, SQL_C_INTERVAL_MONTH, etc.).

### BOOLEAN to interval conversions

- Single-field interval types: TRUE converts to 1, FALSE converts to 0 in the target interval field
- Multi-field interval types: Rejected with SQLSTATE 22015 (interval field overflow)
- Added comprehensive test coverage for all interval types and edge cases

### FLOAT to interval conversions

- Single-field interval types: Float value truncated to integer for the target interval field
- Fractional parts truncated with SQLSTATE 01S07 warning for non-second intervals
- SQL_C_INTERVAL_SECOND preserves microsecond precision in the fraction field
- Multi-field interval types: Rejected with SQLSTATE 22015
- Leading field precision validation enforced (default 2 digits, customizable via SQLSetDescField)
- NaN/Infinity values rejected with SQLSTATE 22003
- Comprehensive test coverage including edge cases and precision handling

### Implementation details

- Refactored numeric helper functions to support both integer and float-based interval conversions
- Added `build_and_write_interval_second` for shared interval second struct creation
- Added `check_leading_precision` and `checked_u32` helper functions for validation
- Updated behavior differences documentation to reflect the new supported conversions

### CODE REVIEW

**Comment by sfc-gh-pbulawa** [comment]
* **#784** <a href="https://app.graphite.com/github/pr/snowflakedb/universal-driver/784?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/snowflakedb/universal-driver/784?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#781** <a href="https://app.graphite.com/github/pr/snowflakedb/universal-driver/781?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: odbc/src/conversion/numeric_helpers.rs [hunk 1]
```diff
@@ -26,6 +26,42 @@ pub fn fractional_warning(has_fractional: bool) -> Warnings {
     }
 }
 
+pub fn check_leading_precision(
+    abs_int: u128,
+    display_value: impl std::fmt::Display,
+    binding: &Binding,
+) -> Result<(), WriteOdbcError> {
+    let leading_precision = binding.datetime_interval_precision.unwrap_or(2) as u32;
+    let exceeds = match 10u128.checked_pow(leading_precision) {
```
Comment (sfc-gh-ddas) [inline odbc/src/conversion/numeric_helpers.rs]: All values that overflow `u128` would pass this check as `checked_pow` would return `None`

File: odbc/src/conversion/boolean_tests.rs [hunk 2]
```diff
@@ -389,6 +389,146 @@ mod tests {
         assert_eq!(buffer[0], 0);
     }
 
+    // ========================================================================
+    // Interval type conversions
+    // ========================================================================
+
+    fn zero_interval() -> sql::IntervalStruct {
```
Comment (sfc-gh-ddas) [inline odbc/src/conversion/boolean_tests.rs]: Duplicate definition. We should extract to a shared header.

File: odbc_tests/BehaviorDifferences.yaml
Comment (sfc-gh-ddas) [inline odbc_tests/BehaviorDifferences.yaml:1]: I think there was a problem during resolving merge conflicts here. Please increment the new BDs after `main`.

**Review by sfc-gh-ddas — CHANGES_REQUESTED** [review]

File: odbc_tests/BehaviorDifferences.yaml
Comment (sfc-gh-pbulawa) [reply odbc_tests/BehaviorDifferences.yaml:1]: Fixed

File: odbc/src/conversion/boolean_tests.rs (hunk 2 above)
Comment (sfc-gh-pbulawa) [reply odbc/src/conversion/boolean_tests.rs]: Fixed

File: odbc/src/conversion/numeric_helpers.rs (hunk 1 above)
Comment (sfc-gh-pbulawa) [reply odbc/src/conversion/numeric_helpers.rs]: Fixed

**Review by sfc-gh-ddas — APPROVED** [review]
